shared/src/stories/eventFiltersModal.stories.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { ChangeEvent, useCallback, useState } from 'react';
+import Modal from '../components/timetable/eventFiltersModal';
+import Button from '../components/common/button';
+import FilterIcon from '../assets/filter.svg?react';
+
+const meta = {
+  title: 'modal/Modal',
+  component: Modal,
+  parameters: {
+    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
+    layout: 'centered',
+  },
+  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
+  tags: ['autodocs'],
+  // More on argTypes: https://storybook.js.org/docs/api/argtypes
+  // Use `fn` to spy on the onClick arg, which will appear in the actions panel once invoked: https://storybook.js.org/docs/essentials/actions#action-args
+} satisfies Meta<typeof Modal>;
+
+export default meta;
+
+export const EventFiltersModal = () => {
+  const [modalActive, setModalActive] = useState(false);
+  const [filters, setFilters] = useState([
+    {
+      id: 'event-type',
+      name: 'Вид мероприятия',
+      values: [
+        { id: 'meeting', value: 'Собрание клуба', isChecked: false },
+        { id: 'concert', value: 'Концерт', isChecked: true },
+        { id: 'festival', value: 'Фестиваль', isChecked: false },
+        { id: 'lecture', value: 'Лекция', isChecked: true },
+        { id: 'event-type-other', value: 'Вид мероприятия', isChecked: false },
+        { id: 'exhibition', value: 'Выставка', isChecked: false },
+      ],
+    },
+    {
+      id: 'audience',
+      name: 'Для кого',
+      values: [
+        { id: 'students', value: 'Для студентов', isChecked: false },
+        { id: 'all', value: 'Для всех', isChecked: false },
+        { id: 'guests', value: 'Для гостей', isChecked: false },
+      ],
+    },
+    {
+      id: 'time',
+      name: 'Когда',
+      values: [
+        { id: 'morning', value: 'Утро', isChecked: false },
+        { id: 'day', value: 'День', isChecked: false },
+        { id: 'evening', value: 'Вечер', isChecked: false },
+      ],
+    },
+    {
+      id: 'status',
+      name: 'Статус',
+      values: [
+        { id: 'pending', value: 'Ожидание', isChecked: false },
+        { id: 'ongoing', value: 'В самом разгаре', isChecked: false },
+        { id: 'past', value: 'Прошедшее', isChecked: false },
+      ],
+    },
+  ]);
+
+  const handleFilterChange = useCallback(
+    (
+      event: ChangeEvent<HTMLInputElement>,
+      filterId: string,
+      valueId: string,
+    ) => {
+      setFilters((prevFilters) =>
+        prevFilters.map((filter) =>
+          filter.id === filterId
+            ? {
+                ...filter,
+                values: filter.values.map((value) =>
+                  value.id === valueId
+                    ? { ...value, isChecked: !value.isChecked }
+                    : value,
+                ),
+              }
+            : filter,
+        ),
+      );
```

Comment:
Нечитаемое выражение. Не стоит для этого использовать .map - он будет проходить по всем элементам
В данном случае можно заменить выражение полностью, но в будущем если будет сложное выражение, которое упростить нельзя, то его следует выделить в другую функцию
```suggestion
      setFilters((prevFilters) => {
        prevFilters
          .find((filter) => filter.id === filterId)!
          .values.find((value) => value.id === valueId)!.isChecked =
          event.target.checked;

        return [...prevFilters];
      });
```